odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/processors/MedicationRequestProcessor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.processors;
+
+import com.ozonehis.eip.odoo.openmrs.handlers.PartnerHandler;
+import com.ozonehis.eip.odoo.openmrs.handlers.SaleOrderHandler;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrder;
+import java.util.List;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.CamelExecutionException;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.ProducerTemplate;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Encounter;
+import org.hl7.fhir.r4.model.Medication;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.openmrs.eip.fhir.Constants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Setter
+@Component
+public class MedicationRequestProcessor implements Processor {
+
+    @Autowired
+    private SaleOrderHandler saleOrderHandler;
+
+    @Autowired
+    private PartnerHandler partnerHandler;
+
+    @Override
+    public void process(Exchange exchange) {
+        try (ProducerTemplate producerTemplate = exchange.getContext().createProducerTemplate()) {
+            Bundle bundle = exchange.getMessage().getBody(Bundle.class);
+            List<Bundle.BundleEntryComponent> entries = bundle.getEntry();
+
+            Patient patient = null;
+            Encounter encounter = null;
+            MedicationRequest medicationRequest = null;
+            Medication medication = null;
+
+            for (Bundle.BundleEntryComponent entry : entries) {
+                Resource resource = entry.getResource();
+                if (resource instanceof Patient) {
+                    patient = (Patient) resource;
+                } else if (resource instanceof Encounter) {
+                    encounter = (Encounter) resource;
+                } else if (resource instanceof MedicationRequest) {
+                    medicationRequest = (MedicationRequest) resource;
+                } else if (resource instanceof Medication) {
+                    medication = (Medication) resource;
+                }
+            }
+
+            if (patient == null || encounter == null || medicationRequest == null || medication == null) {
+                throw new CamelExecutionException(
+                        "Invalid Bundle. Bundle must contain Patient, Encounter, MedicationRequest and Medication",
+                        exchange);
+            } else {
+                log.debug("Processing MedicationRequest for Patient with UUID {}", patient.getIdPart());
```

Comment:
It's good practice for log levels lower than info to check if enabled before logging the message?

Reply:
I think it is configurable in application properties `openmrs.eip.log.level`

Reply:
```suggestion
                 if (log.isDebugEnabled()) log.debug("Processing MedicationRequest for Patient with UUID {}", patient.getIdPart());
```

Reply:
You want me to add `if (log.isDebugEnabled())` condition before every debug log statement. This doesn't seems clean at all.

Reply:
This check helps to avoid the overhead of constructing the log message when the debug level logging is not enabled.

Reply:
It seems like slf4j should take of the if statement.
Have a look here https://stackoverflow.com/a/10428625/12537799

Reply:
Nice that's better.